packages/graphql-auth-transformer/src/resources.ts

```diff
@@ -105,7 +105,8 @@ export class ResourceFactory {
     if (apiKeyConfig && apiKeyConfig.apiKeyExpirationDays) {
       expirationDays = apiKeyConfig.apiKeyExpirationDays;
     }
-    const expirationDateInSeconds = 60 /* s */ * 60 /* m */ * 24 /* h */ * expirationDays; /* d */
+    // add delay expiration time is valid upon resource creation
+    const expirationDateInSeconds = 60 /* s */ * 60 /* m */ * 24 /* h */ * expirationDays /* d */ + 60 * 2;
```

Comment:
I think that we only need to adjust the value if ```expirationDays===1```, otherwise we can run into an issue with 365 days and perhaps the 2 minutes adjustment could be higher like 30-60 minutes to factor in the time that large stack deployments could take.

Also please update the comment to describe why the adjustment is needed.

nth: E2E test for both cases to see that we're not regressing because of this adjustment.